### PR TITLE
HPCC-31074 Refactor ECL Archive Widget to React

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes resize issue with ECL Archive Widget.  Also is first step for direct link to lines of ECL Code (from metrics and error/warning messages).

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith mentioned a couple of small issues. I also added a comment to the Jira with a screenshot about differences I noticed when comparing this branch to a previous version (which would have been 9.4.18 in my case).

**Comment by GordonSmith** [comment]
@jeclrsg - this is ready for review now...

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 1]
```diff
@@ -155,6 +159,9 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
                 <Logs wuid={wuid} filter={queryParams.logs} setLogCount={setLogCount} />
             </DelayLoadedPanel>
             <DelayLoadedPanel visible={tab === "eclsummary"} size={size}>
+                <ECLArchive wuid={wuid} selection={state?.eclsummary} />
+            </DelayLoadedPanel>
+            <DelayLoadedPanel visible={tab === "eclsummaryold"} size={size}>
```
Comment (GordonSmith) [inline esp/src/src-react/components/WorkunitDetails.tsx:164]: Remove "eclsumaryold"

File: esp/src/src-react/util/metricArchive.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        //         fileMetrics.push(...metric.FileMetric.map(fileMetric => {
+        //             return {
+        //                 path: fileMetric.FileName,
+        //                 row: fileMetric.LineNo,
+        //                 col: fileMetric.Column,
+        //                 metric: metric
+        //             };
+        //         }));
+        //     }
+        // });
```
Comment (GordonSmith) [inline esp/src/src-react/util/metricArchive.ts:139]: Remove commented code.

File: esp/src/src-react/hooks/metrics.ts [hunk 3]
```diff
@@ -101,7 +101,21 @@ export enum FetchStatus {
...
 }
 
-export function useWorkunitMetrics(wuid: string): [any[], { [id: string]: any }, WUDetailsMeta.Activity[], WUDetailsMeta.Property[], string[], string[], FetchStatus, () => void] {
+const scopeFilterDefault: WUDetails.RequestNS.ScopeFilter = {
+    MaxDepth: 999999,
+    ScopeTypes: []
+};
+
+const nestedFilterDefault: WUDetails.RequestNS.NestedFilter = {
+    Depth: 999999,
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/hooks/metrics.ts]: Fixed

**Comment by GordonSmith** [comment]
@jeclrsg bump

File: esp/src/src-react/util/metricArchive.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    updateMetrics(metrics: any[]) {
+        this._metrics = metrics;
+        this.sourcePath_TimeTotalExecute = {};
+        this._timeTotalExecute = metrics.filter(metric => !!metric.DefinitionList).reduce((prev, metric) => {
+            const totalTime = metric.TimeMaxTotalExecute ?? metric.TimeAvgTotalExecute ?? metric.TimeTotalExecute ?? 0;
+            metric.DefinitionList?.forEach((definition, idx) => {
+                if (!this.sourcePath_TimeTotalExecute[definition.filePath]) {
+                    this.sourcePath_TimeTotalExecute[definition.filePath] = { total: 0, line: {} };
+                }
+                // if (idx === 0) {
```
Comment (jeclrsg) [inline esp/src/src-react/util/metricArchive.ts:98]: a couple of lines of commented out code here?

File: esp/src/src-react/components/ECLArchiveTree.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    const { ...treeProps } = flatTree.getTreeProps();
+    return <FlatTree {...treeProps} size="small">
+        {
+            Array.from(flatTree.items(), flatTreeItem => {
+                console.log(flatTreeItem.getTreeItemProps());
+                const { fileTimePct, content, ...treeItemProps } = flatTreeItem.getTreeItemProps();
+                return <TreeItem {...treeItemProps} onClick={onClick}>
+                    <TreeItemLayout
+                        // expandIcon={
```
Comment (jeclrsg) [inline esp/src/src-react/components/ECLArchiveTree.tsx:79]: a bit of commented out code

File: esp/src/src-react/components/ECLArchive.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+import { ECLArchiveEditor } from "./ECLArchiveEditor";
+import { MetricsPropertiesTables } from "./MetricsPropertiesTables";
+
+const scopeFilterDefault: WUDetails.RequestNS.ScopeFilter = {
+    MaxDepth: 999999,
+    ScopeTypes: []
+};
+
+const nestedFilterDefault: WUDetails.RequestNS.NestedFilter = {
+    Depth: 999999,
```
Comment (jeclrsg) [inline esp/src/src-react/components/ECLArchive.tsx:21]: Default should be 0?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith two or three small things. Looks good though.

File: esp/src/src-react/components/ECLArchive.tsx (hunk 6 above)
Comment (GordonSmith) [reply esp/src/src-react/components/ECLArchive.tsx:21]: Not here - really it should match:  https://github.com/hpcc-systems/HPCC-Platform/blob/master/esp/src/src/ECLArchiveWidget.ts#L172-L201

I will tweak...

**Comment by GordonSmith** [comment]
@jeclrsg - please re-review

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**